src/v/datalake/translation/deps.h

```diff
@@ -128,6 +128,24 @@ class data_source {
       ss::abort_source&)
       = 0;
 
+    /**
+     * current_lag_ms - Approximation of current translation lag with respect to
+     * some data source.
+     *
+     * Calulated as the difference between current system time and the last time
+     * we reached the topic's high water mark. (i.e. last offset translated was
+     * max_offset_for_translation at the time a translation finished).
+     *
+     * returns optional<chrono::milliseconds>
+     *   - nullopt if iceberg disabled or stm is not in sync
+     *   - 0ms if last_translated_offset == max_offset_for_translation
+     *   - nullopt if we never caught up to max (TODO: not ideal)
```

Comment:
I'm starting to wonder if we should use bytes_remaining as proxy metric for back pressure given this situation  🤔 .. Imagine translating a large partition with historic data and never catching up with the tip of the log and returning nullopt here and the backpressure controller doesn't know what to do, wonder if this case has already been discussed.

Reply:
i think the hand wavy argument was that "never caught up to max" suggests that the partition has high throughput relative to avail resources and should be prioritized, but yea the historic data case is sort of pathological in that sense. besides which that state shouldn't overlap w/ the iceberg disables / stm sync case IMO.

> bytes_remaining

interesting. need to give it a bit more thought. is that readily available and/or cheap to compute?

Reply:
I am thinking more and more about this, is it possible that we will never translate up to the `max_translatable_offset` but still we will be within the boundary of target lag ?

Reply:
Yeah almost certainly